src/resources/embed/index.ts

```diff
@@ -3,12 +3,30 @@ import { RequestOptions } from '../../core';
 import * as Models from '../../models';
 import { APIResource } from '../../resource';
 import { TwelveLabs } from '../..';
-import { CreateEmbedParams, CreateEmbeddingsTaskVideoParams, ListEmbeddingsTaskParams } from './interfaces';
+import { CreateEmbedParams, CreateEmbeddingsTaskVideoParams, ListEmbeddingsTaskParams, RetrieveEmbeddingsTaskParams } from './interfaces';
 import { attachFormFile, removeUndefinedValues, convertKeysToSnakeCase } from '../../util';
 
 export class EmbedTask extends APIResource {
-  async retrieve(id: string, options: RequestOptions = {}): Promise<Models.EmbeddingsTask> {
-    const res = await this._get<Models.EmbeddingsTaskResponse>(`embed/tasks/${id}`, {}, options);
+  async retrieve(
+    id: string,
+    params: RetrieveEmbeddingsTaskParams = {},
+    options: RequestOptions = {}
+  ): Promise<Models.EmbeddingsTask> {
+    let queryString = '';
+
+    if (params.embeddingOption && params.embeddingOption.length > 0) {
+      params.embeddingOption.forEach(option => {
+        queryString += `embedding_option=${encodeURIComponent(option)}&`;
+      });
+      // Remove trailing &
+      queryString = queryString.slice(0, -1);
+    }
+
+    const res = await this._get<Models.EmbeddingsTaskResponse>(
+      `embed/tasks/${id}${queryString ? '?' + queryString : ''}`,
+      {},
+      options
+    );
```

Comment:
```suggestion
    const _params = convertKeysToSnakeCase(params)

    const res = await this._get<Models.EmbeddingsTaskResponse>(
      `embed/tasks/${id}`,
      removeUndefinedValues(_params),
      options
    );
```

in the built-in http client, we don't need to manually parse query string for each param.
fyi) `APIClient`'s `_request` method has this logic:
```ts
if (params) {
  Object.keys(params).forEach((key) => url.searchParams.append(key, params![key]));
}
```

Reply:
hey thanks for suggestion. That unfortunately won't work when multiple array params are passed, which I am not 100% sure why unless I become ts expert. 

error: 
```
/Users/yeonhoo/twelvelabs-js/src/core.ts:125
          throw new Errors.BadRequestError(message, response.status, body, headers, request);
                ^
BadRequestError: The embedding_option[0] parameter is invalid. You should use one of the following values: visual-text, audio.
    at handleResponse (/Users/yeonhoo/twelvelabs-js/src/core.ts:125:17)
    at EmbedTask._request (/Users/yeonhoo/twelvelabs-js/src/core.ts:79:7)
    at processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async EmbedTask._get (/Users/yeonhoo/twelvelabs-js/src/core.ts:98:13)
    at async EmbedTask.retrieve (/Users/yeonhoo/twelvelabs-js/src/resources/embed/index.ts:16:17)
    at async EmbeddingsTask.retrieve (/Users/yeonhoo/twelvelabs-js/src/models/embed/index.ts:126:12)
    at async /Users/yeonhoo/twelvelabs-js/examples/embed.ts:67:10 {
  status: 400,
  body: {
    code: 'parameter_invalid',
    message: 'The embedding_option[0] parameter is invalid. You should use one of the following values: visual-text, audio.'
  },
  headers: Headers {
    [Symbol(map)]: [Object: null prototype] {
      date: [Array],
      'content-type': [Array],
      'content-length': [Array],
      connection: [Array],
      'tl-report': [Array],
      vary: [Array],
      'x-trace-id': [Array],
      'strict-transport-security': [Array]
    }
  },
  request: {
    method: 'GET',
    headers: { 'x-api-key': 'tlk_2P79VSQ1TM4HZC2NBF7SR09RJQ5H' },
    body: undefined
  }

```


This is a solution that I came up with that is more understandable compared to the original. How does this look to you?
```
    const { embeddingOption } = params;

    let url = `embed/tasks/${id}`;
    if (embeddingOption?.length) {
      const queryParams = new URLSearchParams();
      embeddingOption.forEach(option => {
        queryParams.append('embedding_option', option);
      });
      url += `?${queryParams.toString()}`;
    }
```

Reply:
[246d839](https://github.com/twelvelabs-io/twelvelabs-js/pull/101/commits/246d83957e87d0aa4bfe003e9b6350999a1b6eab)

Reply:
I think we can generally address this issue like this:
```ts
// in core.ts
...
if (params) {
  Object.keys(params).forEach((key) => {
    const value = params![key];
    if (Array.isArray(value)) {
      value.forEach((item) => url.searchParams.append(key, item));
    } else {
      url.searchParams.append(key, value);
    }
  });
}
...
```
then just providing params like other methods

Reply:
that is the best, thanks!